ui/src/pages/user_profile_page/UserProfilePage.tsx

```diff
@@ -0,0 +1,108 @@
+import { createSignal, JSXElement, Show } from 'solid-js'
+import { useUser } from '../../context'
+import { clsx } from 'clsx'
+
+import api from '../../api'
+import { ChangePasswordForm } from './ChangePasswordForm'
+import { Modal } from '../../components/Modal'
+
+export function UserProfilePage(): JSXElement {
+  const { user } = useUser()
+  const [openPasswordModal, setOpenPasswordModal] = createSignal(false)
+
+  return (
+    <>
+      <div class="mt-4 ml-10">
+        <Show when={user().isAdmin}>
+          <p class="text-lg text-success mr-4 mb-6 col-span-4">
+            <i class="fa-solid fa-screwdriver-wrench mr-2" />
+            This user is an admin
+          </p>
+        </Show>
+
+        <div>
+          <table class="table">
+            <tbody>
+              <tr>
+                <td>Email:</td>
+                <td>{user().email}</td>
+                <td>
+                  <VerifyEmailButton />
+                </td>
+              </tr>
+              <tr>
+                <td>Password:</td>
+                <td>*********</td>
+                <td>
+                  <p class="tooltip" data-tip="Change password">
+                    <button
+                      class={clsx('btn btn-ghost btn-sm')}
+                      onClick={() => setOpenPasswordModal(true)}
+                    >
+                      <i class="fa-solid fa-edit" />
+                    </button>
+                  </p>
+                </td>
+              </tr>
+            </tbody>
+          </table>
+        </div>
+
+        <ChangePasswordModal
+          isOpen={openPasswordModal()}
+          onClose={() => setOpenPasswordModal(false)}
+        />
+      </div>
+    </>
+  )
+}
+
+function VerifyEmailButton(): JSXElement {
+  const { user } = useUser()
+  const [sending, setSending] = createSignal(false)
+
+  const resendVerificationMail = () => {
+    setSending(true)
+    api.post('/resend_email_verification').finally(() => setSending(false))
+  }
+
+  return (
+    <p
+      class="tooltip"
+      data-tip={user().isVerified ? 'Resend verification' : 'Verify email'}
```

Comment:
This is in the wrong order I think, if the user is verified you want to show something like 'Email verified' else 'Resend verification'